File: BeatPlannerAndroid/src/beatplanner/core/BeatPlanner.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace BeatPlanner
{

  public class Meter
  {
    public int Upper;
    public int Lower;

    public Meter(int u, int l)
    {
      Upper = u;
      if (l == 2 || l == 4 || l == 8 || l == 16 || l == 32)
        Lower = l;
      else
        throw new ArgumentException("Invalid lower meter given");
    }

    public override string ToString()
    {
      return string.Format("{0}/{1}", Upper, Lower);
    }

    public static readonly Meter Common = new Meter(4, 4);
  }

  public class Beat
  {
    public int BPM;
    public Meter Meter;

    public Beat(Meter meter, int BPM)
    {
      this.BPM = BPM;
      this.Meter = meter;
    }

    public static Beat Parse(string[] splits)
    {
      string bpmStr = splits[1];
      int bpm = Convert.ToInt32(bpmStr);

      string meterStr = splits[0];
      string[] meterSplits = meterStr.Split('/');
      int upper = Convert.ToInt32(meterSplits[0]);
      int lower = Convert.ToInt32((meterSplits[1]));

      return new Beat(new Meter(upper, lower), bpm);
    }
    // METER BPM
    // 4/4   180
    public static Beat Parse(string line)
    {
      string[] splits = line.Split(new [] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
      return Parse(splits);

    }
  }


  public class BeatPlanner
  {
    public static void Main(string[] args)
    {
      Track track = Track.Load("assets/track1.txt");
      Metronome metro = new Metronome(
                          new Beat(Meter.Common, 60)
                        );

      TrackPlayer player = new TrackPlayer(track, metro);
      player.PlayAsync();
      String read = "";
      do
      {
        if (read == "stop")
        {
          player.Stop();
        }
        read = Console.ReadLine();
      } while (read != "q");
      player.Stop();
    }

    public static void LoadSaveTest(string[] args)
    {
      Track track = Track.Load("assets/track1.txt");
      TrackPlayer player = new TrackPlayer(track);
      track.Save("assets/track2.txt");
//			player.Play();
    }

    public static void ParseBeatTest(string[] args)
    {
      string input = Console.ReadLine();
      Metronome metro = new Metronome();

      while (input != "q")
      {
        Beat beat = Beat.Parse(input);
        metro.Beat = beat;
        metro.Start();
        while (metro.Bars < 2)
          ;
        metro.Stop();
        metro.Reset();
        input = Console.ReadLine();
      }
    }

    public static void TrackTest(string[] args)
    {
      Track track = new Track();
      TrackPlayer player = new TrackPlayer(track);
      track.AppendSequence(new Beat(Meter.Common, 80), 4);
      track.AppendSequence(new Beat(new Meter(3, 4), 120), 4);
      track.AppendSequence(new Beat(new Meter(7, 8), 200), 4);
      track.PrependSequence(new Beat(new Meter(5, 4), 160), 2);

      player.Play();
    }

    public static void MetroTest(string[] args)
    {
      Console.WriteLine("Hello!");

      Metronome metro = new Metronome();
      // metro.Start();
			

      String read;
      while (true)
      {
        read = Console.ReadLine();
        switch (read)
        {
          case "q":
          case "quit":
            metro.Stop();
            return;

          case "stop":
            metro.Stop();
            break;

          case "start":
            metro.Start();
            break;

          case "reset":
            metro.Reset();
            break;

          case "restart":
            metro.Restart();
            break;

          case "bpm":
            string bpmstr = Console.ReadLine();
            int bpm = Convert.ToInt32(bpmstr);
            metro.Lock.EnterWriteLock();
            metro.BPM = bpm;
            metro.Lock.ExitWriteLock();
						// metro.BPM = bpm;
            break;

          case "beat":
            string beatstr = Console.ReadLine(); // e.g. 3/4
            string[] nums = beatstr.Split('/');
            int upper = Convert.ToInt32(nums[0]);
            int lower = Convert.ToInt32(nums[1]);
						// Console.WriteLine(upper);
						// Console.WriteLine(lower);
            metro.Lock.EnterWriteLock();
            metro.Meter = new Meter(upper, lower);
            metro.Lock.ExitWriteLock();
            break;

          default:
            break;
        }
      }
			
    }
  }
}

File: app/src/main/java/dk/aschoen/beatplanner/todo/BeatPlanner.java
//
// Translated by CS2J (http://www.cs2j.com): 2014-07-29 22:02:11
//

package dk.aschoen.beatplanner.todo;

import dk.aschoen.beatplanner.core.Beat;
import dk.aschoen.beatplanner.core.Meter;
import dk.aschoen.beatplanner.core.Metronome;
//
//@Deprecated
//public class BeatPlanner
//{
//    public static void activity_main(String[] args) throws Exception {
//        BeatPlanner.activity_main(args);
//    }
//
//    public static void activity_main(String[] args) throws Exception {
//        activity_track track = activity_track.load("assets/track1.txt");
//        Metronome metro = new Metronome(new beat(Meter.Common,60));
//        TrackPlayer player = new TrackPlayer(track,metro);
//        player.playAsync();
//        String read = "";
//        do
//        {
//            if (StringSupport.equals(read, "stop"))
//            {
//                player.stop();
//            }
//
//            read = Console.ReadLine();
//        }
//        while (!StringSupport.equals(read, "q"));
//        player.stop();
//    }
//
//    public static void loadSaveTest(String[] args) throws Exception {
//        activity_track track = activity_track.load("assets/track1.txt");
//        TrackPlayer player = new TrackPlayer(track);
//        track.save("assets/track2.txt");
//    }
//
//    //			player.Play();
//    public static void parseBeatTest(String[] args) throws Exception {
//        String input = Console.ReadLine();
//        Metronome metro = new Metronome();
//        while (!StringSupport.equals(input, "q"))
//        {
//            beat beat = beat.parse(input);
//            metro.setBeat(beat);
//            metro.start();
//            while (metro.getBars() < 2)
//                ;
//            metro.stop();
//            metro.reset();
//            input = Console.ReadLine();
//        }
//    }
//
//    public static void trackTest(String[] args) throws Exception {
//        activity_track track = new activity_track();
//        TrackPlayer player = new TrackPlayer(track);
//        track.appendSequence(new beat(Meter.Common,80),4);
//        track.appendSequence(new beat(new Meter(3,4),120),4);
//        track.appendSequence(new beat(new Meter(7,8),200),4);
//        track.prependSequence(new beat(new Meter(5,4),160),2);
//        player.play();
//    }
//
//    public static void metroTest(String[] args) throws Exception {
//        Console.WriteLine("Hello!");
//        Metronome metro = new Metronome();
//        // metro.Start();
//        String read = new String();
//        while (true)
//        {
//            read = Console.ReadLine();
//            System.String __dummyScrutVar0 = read;
//            if (__dummyScrutVar0.equals("q") || __dummyScrutVar0.equals("quit"))
//            {
//                metro.stop();
//                return ;
//            }
//            else if (__dummyScrutVar0.equals("stop"))
//            {
//                metro.stop();
//            }
//            else if (__dummyScrutVar0.equals("start"))
//            {
//                metro.start();
//            }
//            else if (__dummyScrutVar0.equals("reset"))
//            {
//                metro.reset();
//            }
//            else if (__dummyScrutVar0.equals("restart"))
//            {
//                metro.restart();
//            }
//            else if (__dummyScrutVar0.equals("bpm"))
//            {
//                String bpmstr = Console.ReadLine();
//                int bpm = Integer.valueOf(bpmstr);
//                metro.getLock().EnterWriteLock();
//                metro.setBPM(bpm);
//                metro.getLock().ExitWriteLock();
//            }
//            else // metro.BPM = bpm;
//            if (__dummyScrutVar0.equals("beat"))
//            {
//                String beatstr = Console.ReadLine();
//                // e.g. 3/4
//                String[] nums = beatstr.Split('/');
//                int upper = Integer.valueOf(nums[0]);
//                int lower = Integer.valueOf(nums[1]);
//                // Console.WriteLine(upper);
//                // Console.WriteLine(lower);
//                metro.getLock().EnterWriteLock();
//                metro.setMeter(new Meter(upper,lower));
//                metro.getLock().ExitWriteLock();
//            }
//            else
//            {
//            }
//        }
//    }
//
//}


